src/app/src/components/GroupedBarChart.js

```diff
@@ -1,20 +1,40 @@
 import React, { useRef } from 'react';
+import { useSelector } from 'react-redux';
 import { Box } from '@chakra-ui/react';
 import { ResponsiveBarCanvas } from '@nivo/bar';
+import sortBy from 'lodash/sortBy';
+import flatMap from 'lodash/flatMap';
 
 import DownloadMenu from './DownloadMenu';
 import { formatGroupedCSV } from '../utils/csv';
 
+const formatResponsesForChart = ({ items, sortedYears }) => {
+    let data = sortedYears
+        .map(year => {
+            return items
+                .filter(({ response }) => response.year === year)
+                .filter(({ response }) => !response.dataUnavailable)
```

Comment:
Consider combining these two filters in one, to halve the number of times the array has to be traversed

```suggestion
                .filter(({ response }) => response.year === year && !response.dataUnavailable)
```

Reply:
Ah I see you're following the existing style.